common/network-shuffle/src/test/java/org/apache/spark/network/shuffle/TestShuffleDataContext.java

```diff
@@ -97,13 +96,34 @@ public void insertSortShuffleData(int shuffleId, int mapId, byte[][] blocks) thr
 
   /** Creates spill file(s) within the local dirs. */
   public void insertSpillData() throws IOException {
-    String filename = "temp_local_" + UUID.randomUUID();
-    OutputStream dataStream = null;
+    String filename = "temp_local_uuid";
```

Comment:
Yes, but it is called by once for each data context. 

To improve it:
1) an assert can be added or 
2) a counter can be introduced which increased after each call and at the checks this counter can considered when the expected values are constructed

I think the assert should be enough here.

Reply:
What's wrong with the original code you changed?

Reply:
I would like to avoid testing an expression with the same expression: a filter for files with another filter for files. 

Expected values as a literal are much easier to follow.

Reply:
In the old code the test filter was: org.apache.spark.network.shuffle.NonShuffleFilesCleanupSuite#filter:

~~~java
 private static FilenameFilter filter = new FilenameFilter() {
    @Override
    public boolean accept(File dir, String name) {
      // Don't delete shuffle data or shuffle index files.
      return !name.endsWith(".index") && !name.endsWith(".data");
    }
  };
~~~

And the tested code the filter in:
org.apache.spark.network.shuffle.ExternalShuffleBlockResolver#deleteNonShuffleFiles

~~~java
private void deleteNonShuffleFiles(String[] dirs) {
    FilenameFilter filter = new FilenameFilter() {
      @Override
      public boolean accept(File dir, String name) {
        // Don't delete shuffle data or shuffle index files.
        return !name.endsWith(".index") && !name.endsWith(".data");
      }
    };
~~~

Reply:
Ok, since the tests are passing I'll give up on this, but it's weird.